projects/pastanaga/src/lib/dialog/basic-confirm-dialog.component.ts

```diff
@@ -0,0 +1,22 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit, ViewChild } from '@angular/core';
+import { IDialog } from './base-dialog.component';
+import { ConfirmDialogComponent } from './confirm-dialog.component';
+
+@Component({
+    selector: 'pa-basic-confirm-dialog',
+    templateUrl: './basic-confirm-dialog.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BasicConfirmDialogComponent implements IDialog, OnInit {
+    @ViewChild(ConfirmDialogComponent, {static: true}) dialog: ConfirmDialogComponent | undefined;
```

Comment:
Why `dialog: ConfirmDialogComponent | undefined` instead of `dialog?: ConfirmDialogComponent`?

Reply:
`dialog?: ConfirmDialogComponent` means this property is not mandatory, so it can be undefined.
`dialog: ConfirmDialogComponent | undefined` means this property is mandatory but can be undefined

It's subtle, but important: `IDialog` interface is there to make sure any dialog component has a `dialog` property because this dialog property is required by our service to manage opening/closing dialogs properly.